server/src/main/java/org/opensearch/index/shard/StoreRecovery.java

```diff
@@ -463,9 +466,32 @@ private void recoverFromRemoteStore(IndexShard indexShard) throws IndexShardReco
             for (String file : storeDirectory.listAll()) {
                 storeDirectory.deleteFile(file);
             }
+            String segmentInfosSnapshotFilename = null;
             for (String file : remoteDirectory.listAll()) {
                 storeDirectory.copyFrom(remoteDirectory, file, file, IOContext.DEFAULT);
+                if (file.startsWith(SEGMENT_INFO_SNAPSHOT_FILENAME)) {
+                    segmentInfosSnapshotFilename = file;
+                }
             }
+
+            if (segmentInfosSnapshotFilename != null) {
+                String[] filenameTokens = segmentInfosSnapshotFilename.split("__");
+                try (
+                    ChecksumIndexInput indexInput = new BufferedChecksumIndexInput(
+                        storeDirectory.openInput(segmentInfosSnapshotFilename, IOContext.DEFAULT)
+                    )
+                ) {
+                    SegmentInfos infos_snapshot = SegmentInfos.readCommit(
+                        store.directory(),
+                        indexInput,
+                        Integer.parseInt(filenameTokens[1])
+                    );
+                    store.commitSegmentInfos(infos_snapshot, Long.parseLong(filenameTokens[2]), Long.parseLong(filenameTokens[2]));
+                } catch (IOException e) {
+                    logger.info("Exception while reading {}, falling back to commit level restore", segmentInfosSnapshotFilename);
+                }
+            }
```

Comment:
Seems that the handling isn't correct, f.e if we support refresh level durability but are unable to guarantee it should be not be a silent(info logging) failure.

Reply:
Also lets add assertions on checkpoints and max seq no

Reply:
@Bukhtawar I will add the required assertions but wanted to check on the first comment. 

In the first comment, are you suggesting to fail the restore if refresh level durability is not possible? In that case, we need to provide a way to restore from the last commit if user wants. 
Would having a query parameter to the restore API work?